frontend/public/components/utils/useEventListener.jsx

```diff
@@ -0,0 +1,26 @@
+import { useRef, useEffect } from 'react';
+
+const useEventListener = (eventName, handler, element = window) => {
+  const updatedHandler = useRef();
```

Comment:
Why do we need to track `handler` via `useRef` ?

Isn't `handler` supposed to be immutable, i.e. not expected to change over time?

Reply:
why not? a different consumer can choose to change the handler over time